File: core/internal/module/instance/service/service.go
package service

import (
	"github.com/CyaniAgent/Asagity/core/internal/module/instance/dto"
	"github.com/CyaniAgent/Asagity/core/internal/module/instance/model"
	"github.com/CyaniAgent/Asagity/core/internal/module/instance/repository"
	"github.com/CyaniAgent/Asagity/core/internal/platform/config"
)

type Service struct {
	repo repository.Repository
	cfg  config.Config
}

func New(repo repository.Repository, cfg config.Config) *Service {
	return &Service{repo: repo, cfg: cfg}
}

func (s *Service) Version() dto.VersionResponse {
	return dto.VersionResponse{Name: "Asagity Core", Version: "dev"}
}

func (s *Service) Meta() dto.MetaResponse {
	return dto.MetaResponse{
		Name:        "Asagity",
		Alias:       "",
		Description: "Asagity instance metadata is not configured yet.",
	}
}

func (s *Service) GetAllSettings() ([]model.InstanceSetting, error) {
	return s.repo.GetAllSettings()
}

func (s *Service) GetDatabaseStats() ([]repository.DatabaseStat, error) {
	return s.repo.GetDatabaseStats()
}

File: core/src/Verse.Modules/Verse.Module.Instance/Service.cs
using Verse.Shared;

namespace Verse.Module.Instance;

/// <summary>Error codes ported from Go <c>instance/handler</c>.</summary>
public static class InstanceErrors
{
    public const string InternalError = "INTERNAL_ERROR";
}

/// <summary>Contract ported from Go <c>instance/service.Service</c>.</summary>
public interface IInstanceService
{
    VersionResponse Version();
    MetaResponse Meta();
    IReadOnlyList<InstanceSetting> GetAllSettings();
    IReadOnlyList<DatabaseStat> GetDatabaseStats();
    SystemEnvironmentResponse SystemEnvironment();
}

/// <summary>DI stub: every call returns 501 until the real service lands.</summary>
public sealed class NotImplementedInstanceService : IInstanceService
{
    private static ModuleException Todo() =>
        new(StatusCodes.Status501NotImplemented, "INSTANCE_NOT_IMPLEMENTED", "Instance service is not implemented yet.");

    public VersionResponse Version() => throw Todo();
    public MetaResponse Meta() => throw Todo();
    public IReadOnlyList<InstanceSetting> GetAllSettings() => throw Todo();
    public IReadOnlyList<DatabaseStat> GetDatabaseStats() => throw Todo();
    public SystemEnvironmentResponse SystemEnvironment() => throw Todo();
}
